### PR TITLE
fuzz: add .Delete fuzzing

### PR DESCRIPTION
**Pull request by odeke-em**
Adds an oss-fuzz pass for fuzzing just .Delete in which a tree is populated by corpus generated by <KEY>*<VALUE>[...]<DELETE_KEY>

/cc @cuonglm

### CODE REVIEW

File: fuzz/delete/fuzz.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	smn, smv := smt.NewSimpleMap(), smt.NewSimpleMap()
+	tree := smt.NewSparseMerkleTree(smn, smv, sha256.New())
+	for i := 0; i < len(splits)-1; i += 2 {
+		key, value := splits[i], splits[i+1]
+		tree.Update(key, value)
+	}
+
+	deleteKey := splits[len(splits)-1]
+	newRoot, err := tree.Delete(deleteKey)
+
```
Comment (liamsi) [inline fuzz/delete/fuzz.go:29]: nit: seems a bit uncommon to have the err check separated with a newline here:
```suggestion
```

**Review by liamsi — APPROVED** [review]
LGTM. Thanks again!

**Review by cuonglm — APPROVED** [review]

**Comment by cuonglm** [comment]
> LGTM. Thanks again!

Sorry to hjack you here @liamsi, would you mind answering my question in https://github.com/celestiaorg/smt/pull/51#issuecomment-887152900.

Thanks!

File: fuzz/delete/fuzz.go (hunk 1 above)
Comment (odeke-em) [reply fuzz/delete/fuzz.go]: Thank you for the suggestion @liamsi, I've addressed it.

**Comment by liamsi** [comment]
replied here: https://github.com/celestiaorg/smt/pull/51#issuecomment-888975911

**Pull request merged**